src/arcaflow_plugin_sdk/schema.py

```diff
@@ -5008,8 +5008,8 @@ def _validate_not_set(data, object_property: PropertyType, path: typing.Tuple[st
             raise ConstraintException(path, "This field is required")
         if object_property.required_if is not None:
             for required_if in object_property.required_if:
-                if (isinstance(data, dict) and required_if in data) or (
-                    hasattr(data, required_if) and getattr(data, required_if) is None
+                if (isinstance(data, dict) and required_if in data and data[required_if] is not None) or (
```

Comment:
I think it would be valuable to add a comment to the if statement that explains the logic behind the if statement.
Also provide context from the outer code.